pinot-common/src/test/java/com/linkedin/pinot/common/partition/PartitionAssignmentGeneratorTest.java

```diff
@@ -0,0 +1,352 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.common.partition;
+
+import com.google.common.collect.Lists;
+import com.linkedin.pinot.common.config.RealtimeTagConfig;
+import com.linkedin.pinot.common.config.SegmentsValidationAndRetentionConfig;
+import com.linkedin.pinot.common.config.TableConfig;
+import com.linkedin.pinot.common.config.TenantConfig;
+import com.linkedin.pinot.common.utils.LLCSegmentName;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.helix.HelixManager;
+import org.apache.helix.model.IdealState;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.*;
+
+
+public class PartitionAssignmentGeneratorTest {
+
+  private String tableName = "aTableName_REALTIME";
```

Comment:
maybe randomize it so we get different starting points, and invoke the test a few different times?
if you are doing this, always a good idea to System.println() the seed value, so that we can reproduce a failing test case if we want to.
you can add this later if you choose to.

Reply:
Removed all code related to picking specific set of servers and number of servers to be picked. Will add randomization of tests when we add the logic to pick random servers

Reply:
See other comment, You still need to start from a random start point, in case number of servers exceed number of partitions times number of replicas